### DISCUSSION TITLE
Set and verify expiration of .AspNetCore.Cookies

### DISCUSSION THREAD

**Discussion by kayh** [dotnet/yarp#2406] [Q&A] [answered]
We're using YARP to authenticate users using Microsoft Identity and OpenIdConnectDefaults.
As far as I understand it, YARP uses the underlying ASP.NET Core functionality, which is simply configured as described here: https://microsoft.github.io/reverse-proxy/articles/authn-authz.html

Is there a way to set and verify the issued auth cookie expiration server side? 
The data protection configuration is buried deep in the AddAuthentication() extension method and I could not manage to change the cookie name or expiration for instance. All the samples for a "regular" ASP.NET Core app I have found so far do not work with YARP.
Setting the cookie expiration is one part, can this also be verified server side somehow?
Thanks for your thoughts.

**Comment by Tratcher** [comment, marked as answer]
Nothing here should be YARP specific. What issues have you had getting auth samples to work with YARP?  How are you setting the expiration?

**Reply by kayh** [reply]
Sorry for the late reply.
You're right, this is independent of YARP.
I was able to configure everything we need:

services
    .AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme)
    .AddMicrosoftIdentityWebApp(config =>
    {
        ...
    },
    cookieAuthenticationOptions =>
    {
        cookieAuthenticationOptions.ExpireTimeSpan = TimeSpan.FromMinutes(720);
        cookieAuthenticationOptions.SlidingExpiration = false;
        cookieAuthenticationOptions.Cookie.Name = "MyAuthCookieName";
    });
